stellarphot/photometry/photometry.py

```diff
@@ -1,138 +1,408 @@
 
-    # Obtain the local background/pixel and net flux between the aperture and
-    # annulus objects
-    n_pix_ap = apertures.area
-    n_pix_ann = annulus.area
+    try:
+        photom['date-obs'] = Column(data=Time([ccd_image.header['DATE-OBS']]
+                                              * len(photom),
+                                    format='isot', scale='utc'), name='date-obs')
+    except KeyError:
+        raise ValueError("single_image_photometry: 'DATE-OBS' not found in CCD image "
+                         "header ... SKIPPING THIS IMAGE!")
+
+    # Check for airmass keyword in header and set 'airmass' if found,
+    # but accept it may not be available
+    try:
+        photom['airmass'] = [ccd_image.header['AIRMASS']] * len(photom)
+    except KeyError:
+        print("single_image_photometry: 'AIRMASS' not found in CCD "
+                            "image header ... setting to NaN!")
+        photom['airmass'] = [np.nan] * len(photom)
+
+    # Check for filter keyword in header and set 'filter' if found
+    try:
+        photom['filter'] = [ccd_image.header['FILTER']] * len(photom)
+    except KeyError:
+        raise ValueError("single_image_photometry: 'FILTER' not found in CCD "
+                            "image header ... SKIPPING THIS IMAGE!")
+    photom.rename_column('filter', 'passband')
+
+    # Save aperture and annulus information
+    photom.rename_column('aperture_sum_0', 'aperture_sum')
+    photom.rename_column('aperture_sum_1', 'annulus_sum')
+    photom['aperture_sum'].unit = ccd_image.unit
+    photom['annulus_sum'].unit = ccd_image.unit
+    photom['aperture'] = apers.r * u.pixel
+    photom['annulus_inner'] = anuls.r_in * u.pixel
+    photom['annulus_outer'] = anuls.r_out * u.pixel
+    # By convention, area is in units of pixels (not pixels squared) in a digital image
+    photom['aperture_area'] = apers.area * u.pixel
+    photom['annulus_area'] = anuls.area  * u.pixel
 
     if reject_background_outliers:
-        annulus_masks = annulus.to_mask()
-        bkgd_pp = []
-        for mask in annulus_masks:
+        print('single_image_photometry: computing clipped sky stats ...')
+        try:
+            avg_sky_per_pix, med_sky_per_pix, std_sky_per_pix = \
+                    clipped_sky_per_pix_stats(ccd_image, anuls)
+        except AttributeError:
+            print('single_image_photometry: BAD ANNULUS, stats set to np.nan!')
+            avg_sky_per_pix, med_sky_per_pix, std_sky_per_pix = \
+                np.nan, np.nan, np.nan
+        photom['sky_per_pix_avg'] = avg_sky_per_pix * u.pixel**-1
+        photom['sky_per_pix_med'] = med_sky_per_pix * u.pixel**-1
+        photom['sky_per_pix_std'] = std_sky_per_pix * u.pixel**-1
+        print('                          ...DONE computing clipped sky stats')
+    else: # Don't reject outliers (but why would you do this?)
+        print("single_image_photometry: WARNING: Computing sky per pixel without "
```

Comment:
If we want a warning we should using `warning`

Reply:
This got revised in the multi_image_photometry to `suggestion` to avoid that confusion.